### ISSUE TITLE
Add support for Random.generate

### ISSUE THREAD

**Issue by avh4** [avh4/elm-program-test#52]
- [ ] `SimulatedEffect.Random.generate`
- [ ] ProgramTest will just keep a random seed and step it for each generate
- [ ] if there's a ProgramTest.withInitialTime, the initial seed should be generated from that time

**Comment by 7hoenix** [comment]
I was hoping to use elm-program-test to simulate some random generation for a heuristic function (being able to declaratively specify the returned random values sounds really nice for tests).

It looks like this one hasn't made it super far yet though.

Have you started work on it at all?

If not does it seem like something you'd like help with?

**Comment by avh4** [comment]
Yeah, I haven't started on it yet, and I don't think anyone else has either.

With the small note that I'm not gonna merge anything until I'm happy that we have a public API that's easy to use, yeah I'd be happy to have help with this.  (And that would be helpful whether it's help with getting this started, or or help getting it all the way ready to merge.)
